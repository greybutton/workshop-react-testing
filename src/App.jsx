import '@babel/polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import Main from './components/Main';
import cookie from './cookies';

export default (gon) => {
  ReactDOM.render(
    <Main gon={gon} cookie={cookie} />,
    document.getElementById('root'),
  );
};
