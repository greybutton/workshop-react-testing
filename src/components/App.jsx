import React from 'react';
import MyTabs from './Tabs';

const App = () => (
  <div className="container">
    <h1>Hello</h1>
    <MyTabs />
  </div>
);

export default App;
