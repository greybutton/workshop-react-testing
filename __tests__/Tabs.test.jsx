import App from '../src/components/Main';

describe('Tabs', () => {
  const datatest = 'li[data-test^="tab"]';

  test('render', () => {
    const wrapper = mount(<App />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('click on disabled tab', () => {
    const wrapper = mount(<App />);
    const tabDisabled = wrapper.find(datatest).at(1);
    tabDisabled.simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('click on normal tab', () => {
    const wrapper = mount(<App />);
    const tab = wrapper.find(datatest).at(2);
    tab.simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });
});
