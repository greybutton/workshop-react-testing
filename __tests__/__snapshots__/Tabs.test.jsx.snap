// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs click on disabled tab 1`] = `
<div
  class="container"
>
  <h1>
    Hello
  </h1>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-0"
        id="react-tabs-6"
        role="tab"
        tabindex="0"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-9"
        aria-disabled="true"
        aria-selected="false"
        class="react-tabs__tab react-tabs__tab--disabled"
        data-test="tab-1"
        id="react-tabs-8"
        role="tab"
      >
        Title 2 (disabled)
      </li>
      <li
        aria-controls="react-tabs-11"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-2"
        id="react-tabs-10"
        role="tab"
      >
        Title 3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-6"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-7"
      role="tabpanel"
    >
      <h2>
        Any content 1
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-8"
      class="react-tabs__tab-panel"
      id="react-tabs-9"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-10"
      class="react-tabs__tab-panel"
      id="react-tabs-11"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`Tabs click on normal tab 1`] = `
<div
  class="container"
>
  <h1>
    Hello
  </h1>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-13"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-0"
        id="react-tabs-12"
        role="tab"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-15"
        aria-disabled="true"
        aria-selected="false"
        class="react-tabs__tab react-tabs__tab--disabled"
        data-test="tab-1"
        id="react-tabs-14"
        role="tab"
      >
        Title 2 (disabled)
      </li>
      <li
        aria-controls="react-tabs-17"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-2"
        id="react-tabs-16"
        role="tab"
        tabindex="0"
      >
        Title 3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-12"
      class="react-tabs__tab-panel"
      id="react-tabs-13"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-14"
      class="react-tabs__tab-panel"
      id="react-tabs-15"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-16"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-17"
      role="tabpanel"
    >
      <h2>
        Any content 3
      </h2>
    </div>
  </div>
</div>
`;

exports[`Tabs render 1`] = `
<div
  class="container"
>
  <h1>
    Hello
  </h1>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-test="tab-0"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        Title 1
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="true"
        aria-selected="false"
        class="react-tabs__tab react-tabs__tab--disabled"
        data-test="tab-1"
        id="react-tabs-2"
        role="tab"
      >
        Title 2 (disabled)
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-test="tab-2"
        id="react-tabs-4"
        role="tab"
      >
        Title 3
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <h2>
        Any content 1
      </h2>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel"
      id="react-tabs-5"
      role="tabpanel"
    />
  </div>
</div>
`;
